cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/config/ConnectionsConfig.java

```diff
@@ -195,12 +210,56 @@ public CloudEurekaClient primaryEurekaClient(ApplicationInfoManager manager, Eur
         AbstractDiscoveryClientOptionalArgs<?> args = new MutableDiscoveryClientOptionalArgs();
         args.setEurekaJerseyClient(eurekaJerseyClient);
 
-        CloudEurekaClient cloudEurekaClient = new CloudEurekaClient(appManager, config, args,
-            this.context);
+        final CloudEurekaClient cloudEurekaClient = new CloudEurekaClient(appManager, config, args, this.context);
         cloudEurekaClient.registerHealthCheck(healthCheckHandler);
         return cloudEurekaClient;
     }
 
+    @Bean
+    public List<AdditionalRegistration> additionalRegistration(StandardEnvironment environment) {
+        List<AdditionalRegistration> additionalRegistrations = new AdditionalRegistrationParser().extractAdditionalRegistrations(System.getenv());
+        log.debug("Parsed {} additional registrations, \t first: {}", additionalRegistrations.size(), additionalRegistrations.stream().findFirst().orElse(null));
+        return additionalRegistrations;
+    }
+
+    @Bean(destroyMethod = "shutdown")
+    @Conditional(AdditionalRegistrationCondition.class)
+    @RefreshScope
+    public AdditionalEurekaClientsHolder additionalEurekaClientsHolder(ApplicationInfoManager manager,
+                                                                       EurekaClientConfig config,
+                                                                       List<AdditionalRegistration> additionalRegistrations,
+                                                                       EurekaFactory eurekaFactory,
+                                                                       @Autowired(required = false) HealthCheckHandler healthCheckHandler
+    ) {
+        List<CloudEurekaClient> additionalClients = new ArrayList<>(additionalRegistrations.size());
+        for (AdditionalRegistration apimlRegistration : additionalRegistrations) {
+            CloudEurekaClient cloudEurekaClient = registerInTheApimlInstance(config, apimlRegistration, manager, eurekaFactory);
+            additionalClients.add(cloudEurekaClient);
+            cloudEurekaClient.registerHealthCheck(healthCheckHandler);
+        }
+        return new AdditionalEurekaClientsHolder(additionalClients);
+    }
```

Comment:
Honestly, this method seems just like a replacement for processing beans. What about merging this method with `additionalRegistration` and creating just bean `AdditionalEurekaClientsHolder`?

```suggestion
    @Bean(destroyMethod = "shutdown")
    @Conditional(AdditionalRegistrationCondition.class)
    @RefreshScope
    public List<AdditionalRegistration> additionalRegistration(StandardEnvironment environment, ApplicationInfoManager manager,
                                                                       EurekaClientConfig config,
                                                                       List<AdditionalRegistration> additionalRegistrations,
                                                                       EurekaFactory eurekaFactory,
                                                                       @Autowired(required = false) HealthCheckHandler healthCheckHandler) {
        List<AdditionalRegistration> additionalRegistrations = new AdditionalRegistrationParser().extractAdditionalRegistrations(System.getenv());
        log.debug("Parsed {} additional registrations: {}", additionalRegistrations.size(), additionalRegistrations);
        
        return additionalRegistrations.map(apimlRegistration -> {
            CloudEurekaClient cloudEurekaClient = registerInTheApimlInstance(config, apimlRegistration, manager, eurekaFactory);
            cloudEurekaClient.registerHealthCheck(healthCheckHandler);
            return cloudEurekaClient;
        }).collect(Collector.toList());
    }
```

Reply:
discussed